espnet2/asr/frontend/huggingface.py

```diff
@@ -0,0 +1,81 @@
+import copy
+import logging
+from typing import Optional, Tuple, Union
+
+import humanfriendly
+import librosa
+import torch
+from typeguard import typechecked
+
+from espnet2.asr.frontend.abs_frontend import AbsFrontend
+from espnet2.utils.get_default_kwargs import get_default_kwargs
+from espnet.nets.pytorch_backend.frontends.frontend import Frontend
+
+
+class HuggingFaceFrontend(AbsFrontend):
+    """Use pretrained models from HuggingFace for ASR"""
+
+    @typechecked
+    def __init__(
+        self,
+        model,
+        fs: Union[int, str] = 16000,
+        download_dir: Optional[str] = None,
+    ):
+        try:
+            from transformers import AutoFeatureExtractor, AutoModel
+        except ImportError:
+            raise ImportError("Please install `transformers`")
+
+        super().__init__()
+        self.encoder = AutoModel.from_pretrained(model, cache_dir=download_dir)
+        self.processor = AutoFeatureExtractor.from_pretrained(
+            model, cache_dir=download_dir
+        )
+        self.pretrained_params = copy.deepcopy(self.encoder.state_dict())
+
+        if isinstance(fs, str):
+            fs = humanfriendly.parse_size(fs)
+        if fs != self.processor.sampling_rate:
+            logging.warning(
+                f"Sampling rate {fs} does not match upstream model: "
+                + str(self.processor.sampling_rate)
+                + ". Resampling will be performed at forward time."
+            )
+            self.resample = True
+            self.fs = fs
+        else:
+            self.resample = False
+
+    def output_size(self) -> int:
+        return self.encoder.config.hidden_size
+
+    def forward(
+        self, inputs: torch.Tensor, input_lengths: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
```

Comment:
For better understanding, can you add some docstrings about input?

Reply:
Thanks for adding more description! It would be even better to follow this [example](https://github.com/espnet/espnet/blob/master/espnet2/asr/frontend/windowing.py#L57-L66). When a new user of ESPnet or this frontend want to use or extend it, it's more informative and helpful.

Reply:
I reformatted the docstring